diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
index ae0a811..121f892 100644
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
@@ -162,10 +162,11 @@ open class PermissionsService(val context: Context) : InternalModule, Permission
    */
   override fun isPermissionPresentInManifest(permission: String): Boolean {
     try {
-      context.packageManager.getPackageInfo(context.packageName, PackageManager.GET_PERMISSIONS)?.run {
-        return requestedPermissions.contains(permission)
-      }
-      return false
+      val packageInfo = context.packageManager.getPackageInfo(
+        context.packageName,
+        PackageManager.GET_PERMISSIONS
+      )
+      return packageInfo?.requestedPermissions?.contains(permission) ?: false
     } catch (e: PackageManager.NameNotFoundException) {
       return false
     }
