diff --git a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt
index 89d8df9..d0ecb44 100644
--- a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt
+++ b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/safearea/SafeAreaView.kt
@@ -107,8 +107,9 @@ class SafeAreaView(
         val newSystemInsets =
             insets.getInsets(WindowInsetsCompat.Type.systemBars() or WindowInsetsCompat.Type.displayCutout())
 
-        if (newSystemInsets != currentSystemInsets) {
-            currentSystemInsets = EdgeInsets.fromInsets(newSystemInsets)
+        val newEdgeInsets = EdgeInsets.fromInsets(newSystemInsets)
+        if (newEdgeInsets != currentSystemInsets) {
+            currentSystemInsets = newEdgeInsets
 
             if (insetType.containsSystem()) {
                 needsInsetsUpdate = true
